routes/truckRoutes.js

```diff
@@ -0,0 +1,39 @@
+/*
+** author: Fulin Shen
+** created on: May 21, 2016
+** description: see above
+*/
+
+var express = require('express');
+var router = express.Router();
+var bodyParser = require('body-parser');
+var urlEncoded = bodyParser.urlencoded({ extended: false });
+
+var trucks = require('../trucks');
+
+router.route('/')
+  .get(function(request, response) {
+    response.send(trucks.getTrucks());
+  })
+  .post(urlEncoded, function(request, response) {
+    var newTruck = request.body;
+    if(newTruck) {
+      trucks.addTruck(newTruck);
```

Comment:
This still doesn't work as intended, resulting in the added truck appearing as "undefined" in the list until you refresh the page. Here's how I would approach it - first, call `trucks.addTruck(request.body)`. The `addTruck` method in your trucks module should return the thing it added. The return value of addTruck should be assigned to newTruck. Then, send newTruck to the response.